src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -3,12 +3,46 @@
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
+import core.basesyntax.service.exception.InvalidInputDataException;
+import core.basesyntax.service.exception.UserAlreadyExistsException;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final String NULL_INPUT_DATA_MESSAGE = "User data cannot be null";
+    public static final String INVALID_LOGIN_MESSAGE = "User's login is at least 6 characters";
+    public static final String INVALID_PASSWORD_MESSAGE = "User's password is at least 6 characters";
+    public static final String INVALID_MIN_AGE_MESSAGE = "User's age is at least 18 years old";
+    public static final String USER_ALREADY_EXISTS_MESSAGE = "User with the given login already exists";
+    public static final int MIN_LOGIN_LENGTH = 6;
+    public static final int MIN_PASSWORD_LENGTH = 6;
+    public static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUserDataNotNull(user);
```

Comment:
Consider moving the check for existing users (line 25) before the null checks to avoid unnecessary validation if the user already exists.